app/views/unattended/provisioning_templates/snippet/puppet_setup.erb

```diff
@@ -172,14 +150,8 @@ echo "Performing initial full puppet run"
 <% else -%>
 echo "Performing initial puppet run for --tags <%= puppet_tags %>"
 <% end -%>
-<%= bin_path %>/puppet agent --config <%= etc_path %>/puppet.conf --onetime <%= host_param_true?('run-puppet-in-installer') || @full_puppet_run ? '' : "--tags #{puppet_tags}" %> <%= host_puppet_server.blank? ? '' : "--server #{host_puppet_server}" %> --no-daemonize
-<% if os_family == 'Suse' || (os_name == 'Debian' && os_major > 8) || (os_name == 'Ubuntu' && os_major >= 15) -%>
-<% if os_family == 'Suse' -%>
+<%= bin_path %>/puppet agent --config <%= etc_path %>/puppet.conf --onetime <%= host_param_true?('run-puppet-in-installer') || @full_puppet_run ? '' : "--tags #{puppet_tags}" %> --no-daemonize
 <%= bin_path %>/puppet resource service puppet enable=true
-<% else -%>
-systemctl enable puppet
-<% end -%>
-<% end -%>
 <% if @host.provision_method == 'image' -%>
 <%= bin_path %>/puppet resource service puppet ensure=running
```

Comment:
This should add a parameter to the previous command so we only invoke `puppet resource` once.